modules/videoeditor-ffmpeg-impl/src/main/java/org/opencastproject/videoeditor/subtitle/webvtt/WebVTTWriter.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+public class WebVTTWriter {
+  private Charset charset; // Charset used to encode file
+
+  public WebVTTWriter() {
+    this.charset = StandardCharsets.UTF_8;
+  }
+
+  public WebVTTWriter(Charset charset) {
+    this.charset = charset;
+  }
+
+  public void write(WebVTTSubtitle subtitleObject, OutputStream os) throws IOException {
+    try {
+      // Write header
+      List<String> headerLines = subtitleObject.getHeaderLines();
+      for (int i = 0; i < headerLines.size() ; i++) {
+        // Ensure valid header
+        if (i == 0 && !headerLines.get(i).startsWith("WEBVTT")) {
+          os.write(("WEBVTT " +  headerLines.get(i) + "\n").getBytes(this.charset));
+          continue;
+        }
+        os.write((headerLines.get(i) + "\n").getBytes(this.charset));
+      }
+      // Ensure valid header
+      if (headerLines.size() == 0) {
+        os.write(("WEBVTT" + "\n").getBytes(this.charset));
+      }
+      os.write("\n".getBytes(this.charset));
+
+      // Write region blocks
+      for (WebVTTSubtitleRegion region : subtitleObject.getRegions()) {
+        for (String regionLine : region.getLines()) {
+          os.write((regionLine + "\n").getBytes(this.charset));
+        }
+        os.write("\n".getBytes(this.charset));
+      }
+
+      // Write style blocks
+      for (WebVTTSubtitleStyle style : subtitleObject.getStyle()) {
+        for (String styleLine : style.getLines()) {
+          os.write((styleLine + "\n").getBytes(this.charset));
+        }
+        os.write("\n".getBytes(this.charset));
+      }
+
+      // Write cues
+      for (WebVTTSubtitleCue cue : subtitleObject.getCues()) {
+        if (cue.getId() != null) {
+          // Write id
+          String number = String.format("%s\n", cue.getId());
+          os.write(number.getBytes(this.charset));
+        }
+
+        // Write start and end time
+        String startToEnd = String.format("%s --> %s %s\n",
+                this.formatTimeCode(cue.getStartTime()),
+                this.formatTimeCode(cue.getEndTime()),
+                ofNullable(cue.getCueSettingsList()).orElse(""));
+        os.write(startToEnd.getBytes(this.charset));
+
+        // Write text
+        String text = String.format("%s\n", cue.getText());
+        os.write(text.getBytes(this.charset));
+
+        // Write empty line
+        os.write("\n".getBytes(this.charset));
+      }
+    } catch (UnsupportedEncodingException e) {
+      throw new IOException("Encoding error in input subtitle");
+    }
+  }
+
+  private String formatTimeCode(long ms) {
+    long milliseconds = (ms % 1000);
+    long seconds = (long)Math.floor((ms / 1000) % 60);
+    long minutes = (long)Math.floor((ms / (1000 * 60)) % 60);
+    long hours = (long)Math.floor((ms / (1000 * 60 * 60)));
```

Comment:
`Math.floor(…)` does an implicit conversion to a `double` which is why you need to then cast it as a long. But since you only work with `long`, you are already doing an integer division, and you don't need that at all, do you?

```suggestion
    long milliseconds = ms % 1000L;
    long seconds = ms / 1000L % 60L;
    long minutes = ms / (1000L * 60L) % 60L;
    long hours = ms / (1000L * 60L * 60L;
```

Reply:
That does look substantially cleaner, thanks.